crates/spirv-std/src/arch.rs

```diff
@@ -5,6 +5,72 @@
 //! no additional safety checks beyond type-checking.
 use crate::{scalar::Scalar, vector::Vector};
 
+/// Result is true if any component of `vector` is true, otherwise result is
+/// false.
+#[spirv_std_macros::gpu_only]
+#[doc(alias = "OpAny")]
+#[inline]
+pub fn any<V: Vector<bool, N>, const N: usize>(vector: V) -> bool {
+    let mut result = false;
+
+    unsafe {
+        asm! {
+            // Types & Constants
+            "%bool = OpTypeBool",
+            "%uchar = OpTypeInt 8 0",
+            "%uchar_0 = OpConstant %uchar 0",
+            "%uchar_1 = OpConstant %uchar 1",
```

Comment:
Why `uchar` and not `u8` or `byte`?

Reply:
I believe it came from some spirv code from `spirv-dis` but I'll change it. I think u8 makes more sense.